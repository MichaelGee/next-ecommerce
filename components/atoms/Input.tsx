import { Input } from "atomize";


interface IInput {
  placeholder: string;
  prefix: string;
  suffix: string;
  borderFocused: string;
  rounded: string;
  type: string;
}

export const InputComponent = (props: IInput) => {
    const {placeholder, prefix, suffix, borderFocused, rounded, type} = props;
    return (
        <>
         <Input h="2rem" placeholder={placeholder} rounded={rounded} prefix={prefix} suffix={suffix} focusBorderColor={borderFocused} type={type}/>   
        </>
    )
}
