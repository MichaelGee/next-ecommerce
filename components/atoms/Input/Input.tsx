import { Input } from "atomize";


interface IInput {
    placeholder?: string;
    prefix?: React.ReactNode;
    suffix?: React.ReactNode;
    borderFocused?: string;
    borderColor?: string;
    type?: string;
    m?: object;
    p?: object
    shadow?: string;
    rounded?: string;
}

export const InputComponent = (props: IInput) => {
    const { placeholder, prefix, suffix, borderFocused, rounded, type, m, p, shadow, borderColor } = props;
    return (
        <>
            <Input h="2rem" autocomplete="on" m={m} p={p} prefix={prefix} suffix={suffix} placeholder={placeholder} rounded={rounded} focusBorderColor={borderFocused} type={type} shadow={shadow} borderColor={borderColor} />
        </>
    )
}
