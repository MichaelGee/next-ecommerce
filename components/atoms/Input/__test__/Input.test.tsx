import '@testing-library/jest-dom'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { InputComponent } from '../Input';

configure({ adapter: new Adapter() });


describe('render input', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<InputComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('check if input component exist', () => {
        const wrapper = shallow(<InputComponent />);
        expect(wrapper.find(<input />));
    })
})