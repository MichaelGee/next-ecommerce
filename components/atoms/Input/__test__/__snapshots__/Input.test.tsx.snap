// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input matches the snapshot 1`] = `
<div
  className=""
>
  <input
    autocomplete="on"
    className=""
    d="flex"
  />
</div>
`;
